### PR TITLE
[PWGJE] Add efficiency and response matrix (MC)

### PR DESCRIPTION
**Pull request by joachimckh**
I've added process function for matched part and det MC to create the response matrix, and also a process function to fill jet pt to create efficiencies.

### CODE REVIEW

File: PWGJE/Tasks/jetSpectraEseTask.cxx [hunk 1]
```diff
@@ -130,9 +148,72 @@ struct JetSpectraEseTask {
...
+    registry.fill(HIST("h_mc_collisions"), counter++);
+    for (const auto& mcdjet : mcdjets) {
+      if (!mcdjet.has_matchedJetGeo())
+        continue;
+
+      registry.fill(HIST("h_detector_jet_pt"), mcdjet.pt());
+      registry.fill(HIST("h_detector_jet_eta"), mcdjet.eta());
+      registry.fill(HIST("h_detector_jet_phi"), mcdjet.phi());
+      for (auto& mcpjet : mcdjet.template matchedJetGeo_as<JetMCPTable>()) {
+        if (!mcpjet.has_matchedJetGeo())
```
Comment (nzardosh) [inline PWGJE/Tasks/jetSpectraEseTask.cxx:183]: why do you check both directions?

Comment (nzardosh) [inline PWGJE/Tasks/jetSpectraEseTask.cxx:183]: why do you check both directions?

File: PWGJE/Tasks/jetSpectraEseTask.cxx [hunk 2]
```diff
@@ -130,9 +148,72 @@ struct JetSpectraEseTask {
...
+    registry.fill(HIST("h_mc_collisions"), counter++);
+  }
+  PROCESS_SWITCH(JetSpectraEseTask, processMCChargedMatched, "jet matched mcp and mcd", false);
+
+  template <typename T>
+  bool isAcceptedLeadTrack(T const& tracks)
+  {
+    float leadingTrackpT{0.0};
+    for (const auto& track : tracks) {
+      if (track.pt() > 5.0) {
```
[the remark this answers is not in the document]
Comment (joachimckh) [reply PWGJE/Tasks/jetSpectraEseTask.cxx]: Changed this now

File: PWGJE/Tasks/jetSpectraEseTask.cxx [hunk 3]
```diff
@@ -130,9 +148,72 @@ struct JetSpectraEseTask {
...
+        if (track.pt() > leadingTrackpT) {
+          leadingTrackpT = track.pt();
+        }
+      }
+    }
+    if (leadingTrackpT == 0.0)
```
[the remark this answers is not in the document]
Comment (joachimckh) [reply PWGJE/Tasks/jetSpectraEseTask.cxx:209]: I used double instead now

File: PWGJE/Tasks/jetSpectraEseTask.cxx (hunk 1 above)
Comment (joachimckh) [reply PWGJE/Tasks/jetSpectraEseTask.cxx]: I removed the two checks

**Comment by joachimckh** [comment]
@nzardosh There was some git issues from my side. It should be fixed now and the PR should be ready to merge.
I added what your comments suggested, and the final commit was to change the qvector implementation and rid myself of FFitWeights dependency.

**Review by nzardosh — APPROVED** [review]

**Pull request merged**